.github/workflows/build.yml

```diff
@@ -31,18 +29,115 @@ jobs:
           path: .gradle
           key: ${{ runner.os }}-gradle--${{ hashFiles('**/build.gradle', '**/settings.gradle', '**/gradle.properties') }}
 
-      - name: Format with spotless
+      - name: Build with Gradle
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: spotlessApply --build-cache
+          arguments: build --build-cache
 
-      - uses: stefanzweifel/git-auto-commit-action@v4
+      - name: Upload build
+        uses: actions/upload-artifact@v3
         with:
-          commit_user_name: 'Wynntils'
-          commit_user_email: 'admin@wynntils.com'
-          commit_message: 'chore: spotless formatting'
+          name: build
+          path: |
+            forge/build/libs/wynntils-*.MC1.18.2-forge.jar
+            fabric/build/libs/wynntils-*.MC1.18.2-fabric.jar
+            quilt/build/libs/wynntils-*.MC1.18.2-quilt.jar
 
-      - name: Build with Gradle
-        uses: gradle/gradle-build-action@v2
+  changelog:
+    name: Generate Changelog
+    needs: [build]
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.changelog.outputs.tag }}
+      skipped: ${{ steps.changelog.outputs.skipped }}
+      clean_changelog: ${{ steps.changelog.outputs.clean_changelog }}
+      changelog: ${{ steps.changelog.outputs.changelog }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.PRIVATE_TOKEN }}
+
+      - name: Create changelog
+        id: changelog
+        uses: TriPSs/conventional-changelog-action@v3
         with:
-          arguments: build -x spotlessCheck --build-cache
+          github-token: ${{ secrets.PRIVATE_TOKEN }}
+          preset: conventionalcommits
+          release-count: 1
+          pre-commit: .pre-commit.js
+          create-summary: true
+          version-file: ./version.json
+          skip-git-pull: true
+
+  publish-gh:
+    name: Publish to Github Releases
+    runs-on: ubuntu-latest
+    needs: [ changelog, artifact ]
+    steps:
+      - uses: actions/download-artifact@v3
```

Comment:
Probably will need a name for the artifact download

Reply:
I used the following https://github.com/actions/download-artifact#download-all-artifacts